lib/src/flutter_search_bar_base.dart

```diff
@@ -132,7 +136,18 @@ class SearchBar {
   ///
   AppBar buildSearchBar(BuildContext context) {
     ThemeData theme = Theme.of(context);
-    Color buttonColor = inBar ? null : theme.iconTheme.color;
+    TextStyle hintStyle = TextStyle(
+      color: inBar
+          ? theme.appBarTheme?.textTheme?.headline4?.color ?? theme.primaryTextTheme?.headline4?.color
+          : theme.textTheme.headline4.color,
+    );
+    TextStyle inputStyle = inBar ? theme.appBarTheme?.textTheme?.subtitle1 ?? theme.primaryTextTheme?.subtitle1 : null;
+    Color buttonColor = inBar ? inputStyle?.color : theme.iconTheme.color;
```

Comment:
Are we always guaranteed that `theme.iconTheme` will be non-null?

Reply:
Addressed